spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -173,12 +173,22 @@ protected void sendMessage(Message<?> message) {
 		catch (RuntimeException e) {
 			MessageChannel errorChannel = getErrorChannel();
 			if (errorChannel != null) {
-				this.messagingTemplate.send(errorChannel, new ErrorMessage(e));
+				this.messagingTemplate.send(errorChannel, buildErrorMessage(e));
```

Comment:
Ha-ha! OK. I give up on this stuff.
Is there some work which you really can't do and I am still would be useful?  😉